janMarkusMilan/assignments/reverseLinked/lc92/lc92.js

```diff
@@ -0,0 +1,184 @@
+// Problem Statement #
+
+// Given the head of a LinkedList and two positions ‘p’ and ‘q’, reverse the LinkedList from position ‘p’ to ‘q’.
+
+/*
+Insights:
+  - At some point, we need to implement the steps to reverse a linked list
+  - We can just iterate through the list until we reach 'p'
+  - Then we have a loop that will reverse that specific part of the linked list
+  - So, we need a counter and we reverse the linked list once its between p and q
+
+Reversing a Linked List;
+  1. Get current node and store next value
+  2. Update next value to a "listSoFar"
+  3. Point the "listSoFar" to the current node
+  4. Update the current node to store the next value
+
+Thoughts:
+  - We should keep track of the node at p - 1, as it represents the last list node before our reversed section
+  - The first node in the reversed section (pre-reversal) will become the tail
+  - If we implement the logic to reverse a linked list, then our "listSoFar" variable will store the head of the entire reversed linked list section
+*/
+
+class Node {
+  constructor(value, next = null) {
+    this.value = value;
+    this.next = next;
+  }
+
+  get_list() {
+    let result = "";
+    let temp = this;
+    while (temp !== null) {
+      result += temp.value + " ";
+      temp = temp.next;
+    }
+    return result;
+  }
+}
+
+const reverse_sub_list = function (head, p, q) {
+  // TODO: Write your code here
+  let currentPos = 1;
+  let currentNode = head;
+  let start = head; // Represents the last list node before the start of our reversed linked list section
+
+  // Iterate through until we reach p
+  while (currentPos < p) {
+    start = currentNode; // Will end up pointing to the node before the reversed linked list section, or p - 1
+    currentNode = currentNode.next; // Advance current node (will end up equal to value at node p)
+    currentPos++; // Increment position (will end up equal to p)
+  }
+
+  let newList = null; // Initialize a new list
+  let tail = currentNode; // The beginning node in the target section of the linked list will end up as the tail once reversed
+
+  // Now iterate through the reversal while the currentPos is in the boundaries of p and q
+  while (currentPos >= p && currentPos <= q) {
+    const nextNode = currentNode.next;
+    currentNode.next = newList;
+    newList = currentNode; // currentNode will represent the head of the list we have built so far
+    currentNode = nextNode; // At the end, currentNode will equal the value at q + 1, which represents the start of the rest of the linked list
+    currentPos++;
+  }
+
+  start.next = newList; // The newList is the head of the reversed section
+  tail.next = currentNode; // currentNode is pointing at q + 1
+
+  // Most of the time, we could return head, but it won't work if p is 1, meaning we reverse starting from the beginning, so we need to check
+  if (p > 1) {
+    return head;
+  } else {
+    return newList;
+  }
+};
```

Comment:
Very opinionated topic, but, many will say removing the inline comments from codebase is best practice before submitting